src/vm/object.cpp

```diff
@@ -1525,21 +1525,21 @@ void STDCALL CopyValueClassUnchecked(void* dest, void* src, MethodTable *pMT)
     switch (pMT->GetNumInstanceFieldBytes())
     {        
     case 1:
-        VolatileStore((UINT8*)dest, *(UINT8*)src);
+        *(UINT8*)dest = *(UINT8*)src;
         break;
 #ifndef ALIGN_ACCESS
         // we can hit an alignment fault if the value type has multiple 
         // smaller fields.  Example: if there are two I4 fields, the 
         // value class can be aligned to 4-byte boundaries, yet the 
         // NumInstanceFieldBytes is 8
```

Comment:
These triggered alignment faults.  I believe they do not belong here.  There seems to be similar barriers in `FieldDesc::GetInstanceField` and `FieldDesc::SetInstanceField`.  I believe these are also wrong.  They might belong in the JIT helpers that may call them if and only if IL VOLATILE is set.